src/odemis/gui/cont/settings.py

```diff
@@ -838,7 +838,24 @@ def _onUpdateTriggerDelayMD(self, evt):
         requested_triggerDelay = self.ctrl_triggerDelay.GetValue()
         # get a copy of  MD
         trigger2delay_MD = self.streak_delay.getMetadata()[model.MD_TIME_RANGE_TO_DELAY]
-        trigger2delay_MD[cur_timeRange] = requested_triggerDelay
+
+        # check if key already exists (prevent creating new key due to floating point issues)
+        key = util.find_closest(cur_timeRange, trigger2delay_MD.keys())
+        if util.almost_equal(key, cur_timeRange):
+            # Replace the current delay value with the requested for an already existing timeRange in the dict.
+            # This avoid duplication of keys, which are only different because of floating point issues.
+            trigger2delay_MD[key] = requested_triggerDelay
+        else:
+            trigger2delay_MD[cur_timeRange] = requested_triggerDelay
+            logging.warning("A new time range %s was added to the MD, which is not consistent with the timeRanges"
+                            " specified in the yaml file.", cur_timeRange)
+
+        # check the number of keys in the dict is same as choices for VA
+        if len(trigger2delay_MD.keys()) != len(self.streak_unit.timeRange.choices):
+            logging.warning("The number of %s time ranges in the trigger delay MD does not match with the number of %s "
+                            "choices specified in the yaml file.",
```

Comment:
In general, I'd suggest to avoid talking of "the yaml file" in the GUI. Firstly, it's not very precise (as YAML is a just file format), so "microscope file" would be more adapted. More theoretically, I think it's better to keep it generic, and not have expection from the GUI on how the back-end is started.
Anyway, here I think here the time ranges are from the `timeRange.choices`, so you could write something like:
`"MD_TIME_RANGE_TO_DELAY has %d entries, while the device .timeRange has %d choices."`

(also update the message just before).